### PR TITLE
Initial commit for Linkerd RFC proposal process

### PR DESCRIPTION
**Pull request by admc**
@olix0r @grampelberg - WIP please review and comment.

### CODE REVIEW

File: README.md
Comment (grampelberg) [inline README.md]: I like the way that k8s describes this:

> The definition of what constitutes an "enhancement" is a foundational concern for the Kubernetes project. Roughly any Kubernetes user or operator facing enhancement should follow the KEP process: if an enhancement would be described in either written or verbal communication to anyone besides the KEP author or developer then consider creating a KEP.

> Similarly, any technical effort (refactoring, major architectural change) that will impact a large section of the development community should also be communicated widely. The KEP process is suited for this even if it will have zero impact on the typical user or operator.

Comment (grampelberg) [reply README.md]: The workflow I'm imagining in my mind is an issue on linkerd2 first, and the community then asks for an RFC based off potential sizing.

File: README.md
Comment (grampelberg) [inline README.md]: #contributors

File: README.md
Comment (grampelberg) [inline README.md]: This feels pretty fuzzy. We should either reduce or increase the details here.

File: README.md
Comment (grampelberg) [inline README.md]: This seems especially long for us?

File: README.md
Comment (grampelberg) [inline README.md]: `as a series of pull requests`

File: README.md
Comment (grampelberg) [inline README.md]: WDYT about adding an implementation owner to the top? When someone starts working on it, that gets updated with their name.

File: active/linkerd-rfc-process.md
Comment (grampelberg) [inline active/linkerd-rfc-process.md]: to understand and communicate upcoming changes to Linkerd.

The purpose of the RFC process is to reduce the amount of "tribal knowledge" in our community. By moving decisions into a well tracked artifact, this process aims to enhance communication and discoverability.

File: active/linkerd-rfc-process.md
Comment (grampelberg) [inline active/linkerd-rfc-process.md]: to contributors.

File: active/linkerd-rfc-process.md
Comment (grampelberg) [inline active/linkerd-rfc-process.md]: We will be using [the RFC repository](https://github.com/linkerd/rfc) to store

File: active/linkerd-rfc-process.md
Comment (grampelberg) [inline active/linkerd-rfc-process.md]: Reference KEP and rust RFC process?

File: active/linkerd-rfc-process.md
Comment (grampelberg) [inline active/linkerd-rfc-process.md]: - How reviewers and approvers are assigned to an RFC
- Example schedule, deadline, and time frame for each stage of an RFC
- Communication/notification mechanisms
- Review meetings and escalation procedure

File: rfc-template.md
Comment (grampelberg) [inline rfc-template.md]: - Use Cases
- Goals
- Non-Goals

File: rfc-template.md
Comment (grampelberg) [inline rfc-template.md]: I think we need separate proposal and design detail sections.

File: rfc-template.md
Comment (grampelberg) [inline rfc-template.md]: I would like to see a deliverables section, but that could be a sub-heading under design details.

File: rfc-template.md
Comment (grampelberg) [inline rfc-template.md]: Do we need this section? Feels like it should be covered in the other sections as supporting data.

File: rfc-template.md
Comment (grampelberg) [inline rfc-template.md]: This feels a little weird.

**Review by grampelberg — COMMENTED** [review]
Mostly wording suggestions to make this feel more like a process for everyone. The structural suggestions are around the template itself.

Basically, I'd like to split the `Reference-level explanation` into two separate sections that can be filled out at different times during the process (proposal and design details).

The only other large change would be introducing deliverables. I feel it is critical that these RFCs are separated out into different deliverables that are reviewed and merged as distinct pieces. While it isn't *required* that this is all planned up front, a solid RFC should definitely have an idea of how to do the delivery as part of defining the design details succinctly.

File: rfc-template.md
Comment (hawkw) [reply rfc-template.md]: 👍 for "non-goals", it's good to state this clearly.

File: rfc-template.md
Comment (hawkw) [reply rfc-template.md]: Could be a subsection of a "design" section?

File: rfc-template.md
Comment (hawkw) [inline rfc-template.md]:
```suggestion
If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other software.
```
(i'm guessing this sentence came from the Rust RFC process?)

File: rfc-template.md
Comment (hawkw) [reply rfc-template.md]: The current issue template for feature requests has an "alternatives", which I also think is often useful...perhaps these could be combined?

File: README.md
Comment (admc) [reply README.md]: You mean have that as a field at the top of the RFC? Works for me.

Comment (grampelberg) [reply README.md]: Yeah, I was trying to come up with a way that folks could easily see what's available to work on.

File: README.md
Comment (olix0r) [inline README.md]: @admc What are your thoughts on the directory layout in this repo? Will we add an `accepted/` directory where active RFCs are moved?

File: README.md
Comment (olix0r) [inline README.md]: How would you feel about requiring a numbering scheme like `0000-linkerd-rfc-process.md`? Or perhaps `0000/rfc.md`? Basically, What if it makes sense to have auxiliary files? How do they get grouped together? Additionally, it would be nice to have some natural sorting and uniqueness.

File: README.md
Comment (admc) [reply README.md]: removed

File: active/linkerd-rfc-process.md
Comment (admc) [reply active/linkerd-rfc-process.md]: Agreed, added them in the unresolved questions area incase we want that all in this process doc before merge.

File: README.md
Comment (admc) [reply README.md]: I separated the process into two steps, I don't have a strong opinion, but it seems like for a first try, having the whole conversation in one file will be easy to manage.

File: README.md
Comment (admc) [reply README.md]: Added the RFC numbering schema.

**Review by grampelberg — APPROVED** [review]

**Review by olix0r — APPROVED** [review]
I think this lays out all of the top level concerns. I, personaly, don't think I'd break up an RFC into discrete steps -- i'd just post the thing minus implementation plan and let someone else figure that out. But, I think what we have here is directionally close enough that we should roll forward and refine it as we get experience.

**Pull request merged**
